iterative/utils/analytics.go

```diff
@@ -0,0 +1,290 @@
+	"os/exec"
+	"reflect"
+	"regexp"
+	"runtime/debug"
+	"sync"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/shirou/gopsutil/host"
+	"github.com/sirupsen/logrus"
+	"github.com/wessie/appdirs"
+
+	"golang.org/x/crypto/scrypt"
+)
+
+const (
+	Timeout  = 5 * time.Second
+	Endpoint = "https://telemetry.cml.dev/api/v1/s2s/event?ip_policy=strict"
+	Token    = "s2s.jtyjusrpsww4k9b76rrjri.bl62fbzrb7nd9n6vn5bpqt"
+)
+
+var (
+	Version string = "0.0.0"
+	wg      sync.WaitGroup
+)
+
+func getenv(key, defaultValue string) string {
+	value := os.Getenv(key)
+	if len(value) == 0 {
+		return defaultValue
+	}
+	return value
+}
+
+func deterministic(data string) (*uuid.UUID, error) {
+	ns := uuid.NewSHA1(uuid.NameSpaceDNS, []byte("iterative.ai"))
+
+	seed, err := ns.MarshalBinary()
+	if err != nil {
+		return nil, err
+	}
+
+	dk, err := scrypt.Key([]byte(data), seed, 1<<16, 8, 1, 8)
+	if err != nil {
+		return nil, err
+	}
+
+	id := uuid.NewSHA1(ns, []byte(hex.EncodeToString(dk)))
+	return &id, nil
+}
+
+func SystemInfo() map[string]interface{} {
+	hostStat, _ := host.Info()
+	return map[string]interface{}{
+		"platform":         hostStat.Platform,
+		"platform_version": hostStat.PlatformVersion,
+	}
+}
+
+func TaskDuration(logs string) float64 {
+	regex := regexp.MustCompile(`\d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2}`)
+	matches := regex.FindAllString(logs, -1)
+	taskDuration := 0.0
+
+	if len(matches) > 1 {
+		layout := "2006-01-02 15:04:05"
+		t1, _ := time.Parse(layout, matches[len(matches)-1])
+		t0, _ := time.Parse(layout, matches[0])
+		taskDuration = t1.Sub(t0).Seconds()
+	}
+
+	return taskDuration
+}
+
+func IsCI() bool {
+	if _, ok := os.LookupEnv("CI"); ok {
+		return true
+	}
+
+	return false
+}
+
+func guessCI() string {
+	if _, ok := os.LookupEnv("GITHUB_SERVER_URL"); ok {
+		return "github"
+	}
+
+	if _, ok := os.LookupEnv("CI_SERVER_URL"); ok {
+		return "gitlab"
+	}
+
+	if _, ok := os.LookupEnv("BITBUCKET_WORKSPACE"); ok {
+		return "bitbucket"
+	}
+
+	return ""
+}
+
+func TerraformVersion() string {
+	var out bytes.Buffer
+	cmd := exec.Command("terraform", "--version")
+	cmd.Stdout = &out
+
+	err := cmd.Run()
+
+	if err == nil {
+		regex := regexp.MustCompile(`v\d+\.\d+\.\d+`)
+		version := regex.FindString(out.String())
+		return version
+	} else {
+		logrus.Error("Analytics: Failed extracting terraform version")
+	}
+
+	return ""
+}
+
+func GroupId() string {
+	if !IsCI() {
+		return ""
+	}
+
+	id, err := deterministic(fmt.Sprintf("%s%s%s%s%s",
+		os.Getenv("GITHUB_SERVER_URL"),
+		os.Getenv("GITHUB_REPOSITORY_OWNER"),
+		os.Getenv("CI_SERVER_URL"),
+		os.Getenv("CI_PROJECT_ROOT_NAMESPACE"),
+		os.Getenv("BITBUCKET_WORKSPACE"),
+	))
+
+	if err != nil {
+		return ""
+	}
+
+	return id.String()
+}
+
+func UserId() string {
+	id := uuid.New().String()
+	old := appdirs.UserConfigDir("dvc/user_id", "iterative", "", false)
+	new := appdirs.UserConfigDir("telemetry", "iterative", "", false)
```

Comment:
```suggestion
	new := appdirs.UserConfigDir("iterative/user_id", "", "", false)
```

...or something; the specification has changed.

Reply:
Also, @casperdcl, this file has JSON inside, right?

Reply:
```suggestion
	new := appdirs.UserConfigDir("iterative/telemetry", "", "", false)
```

Reply:
> this file has JSON inside, right?

no, just the id I think.

Reply:
But the DVC file contains `{"user_id": "00···0"}` and we're just copying it over the new one. 🤔

Reply:
Should we parse the JSON of the old format and write only the raw value to the new format?

Reply:
I do not follow that @0x2b3bfa0 
DO you say that we should try to decode the JSON (if any) and then write the raw id as it is right now?

Reply:
👍🏼